server/main.py

```diff
@@ -17,14 +22,65 @@ def insert_result_sync(results):
 
 @app.task(name='process_alignment', queue='server_default')
 def process_alignment(data):
-    logger.info(data)
-    logger.info(data['db'])
-    alignment = [{data['net1']: f'node_{i}', data['net2']: f'node_{i}'}
-                 for i in range(100)]
-    data['alignment'] = alignment
-    results = {'results': data}
-    result_id = insert_result_sync(results)
-    send_finished_job(data['job_id'], result_id)
+    logger.info(f'processing alignment {data}')
+
+    job_id = data['job_id']
+
+    db_name = data['db'].lower()
+    net1_name = data['net1']
+    net2_name = data['net2']
+    aligner_name = data['aligner'].lower()
+
+    if db_name == 'isobase':
+        db = sources.IsobaseLocal('/opt/networks/isobase')
+        net1 = db.get_network(net1_name)
+        net2 = db.get_network(net2_name)
+        net1_net2_scores = db.get_score_matrix(net1_name, net2_name, net1=net1, net2=net2)
+
+        if aligner_name == 'alignet':
+            net1_scores = db.get_score_matrix(net1_name)
+            net2_scores = db.get_score_matrix(net2_name)
+            run_args = (net1, net2, net1_scores, net2_scores, net1_net2_scores)
+            aligner = aligners.Alignet()
+        else:
+            run_args = (net1, net2, net1_net2_scores)
+
+            if aligner_name == 'hubalign':
+                aligner = aligners.Hubalign()
+            elif aligner_name == 'lgraal':
+                aligner = aligners.LGraal()
+            elif aligner_name == 'pinalog':
+                aligner = aligners.Pinalog()
+            elif aligner_name == 'spinal':
+                aligner = aligners.Spinal()
```

Comment:
I think this can be more clean with a dispatcher
```
aligners_dispatcher = {'hubalign': Hubalign(), 'lgraal': LGraal(), 'pinalog': Pinalog(), 'spinal': Spinal()
aligner = aligners_dispatcher.get(aligner_name)`
if aligner is None:
    raise ValueError()
```

Reply:
I'll change that. That chain of if's is for future (individual) treatment of individual aligner parameters, but now that I think of it, that could simply be **kwargs and have the dispatcher be
```python
aligners_dispatcher = {'hubalign': Hubalign, 'lgraal': LGraal, 'pinalog': Pinalog, 'spinal': Spinal}
aligner_con = aligners_dispatcher.get(aligner_name)`
if aligner_con is None:
    raise ValueError()
aligner = aligner_con(**aligner_params)
```